polaris-patterns/src/components/PolarisPatternsProvider/PolarisPatternsProvider.tsx

```diff
@@ -0,0 +1,57 @@
+import type {PropsWithChildren} from 'react';
+import React, {createContext, useEffect} from 'react';
+import {I18nextProvider, initReactI18next} from 'react-i18next';
+import {createInstance} from 'i18next';
+
+import {DEFAULT_I18N_DETAILS} from '../../configure';
+
+export const I18nDetails = createContext(DEFAULT_I18N_DETAILS);
+
+const i18next = createInstance({
+  fallbackLng: DEFAULT_I18N_DETAILS.locale,
+  interpolation: {
+    escapeValue: false,
+  },
+  resources: {},
+  returnNull: false,
+});
+
+i18next.use(initReactI18next).init();
+
+interface TranslationDictionary {
+  [key: string]: string | TranslationDictionary;
+}
+
+interface I18nDetails {
+  locale: string;
+  country?: string;
+  currency?: string;
+  timezone?: string;
+}
+
+export interface Props {
+  i18nDetails: I18nDetails;
+  translations: TranslationDictionary;
+  enFallback: TranslationDictionary;
+}
+
+export function PolarisPatternsProvider({
+  translations,
+  i18nDetails,
+  enFallback,
+  children,
+}: PropsWithChildren<Props>) {
+  useEffect(() => {
+    i18next.addResourceBundle('en', 'translation', enFallback);
```

Comment:
You might want to look into the `deep` and `overwrite` options for [`addResourceBundle()`](https://www.i18next.com/overview/api#addresourcebundle) (perhaps for a future PR where `PolarisPatternsProvider` is providing its own translations if they are not provided).

Basically, you need to decide, what do you want the overwrite behaviour to be when someone provides their own `enFallback` translations? The wording for the `deep` option is confusing to me, but I think if you set `deep` to `true` and `overwrite` to `true`, then the translations for `enFallback` will overwrite matching existing translations, but will maintain existing translations that were not provided in `enFallback`, which I think is what you might want here.

```suggestion
    i18next.addResourceBundle('en', 'translation', enFallback, true, true);
```

Reply:
I think I might remove this prop anyways but (see comment below) but maybe add the logic in to overwrite the default english fallback. Thanks!

Reply:
I tried all permutations of those last two boolean params but they didn't seem to make a difference, it was acting as described for `true, true` every time. Either way I added them in to be safe

Reply:
Oh interesting, I haven't tried them before.